### ISSUE TITLE
distributionPath property not honored when caching distributions

### ISSUE THREAD

**Issue by cloudshiftchris** [gradle/gradle-build-action#484]
The `distributionPath` property in `gradle/wrapper/gradle-wrapper.properties` is not used when caching distributions.  It appears that caching is only supported for the default `wrapper/dists` location.

**Comment by bigdaz** [comment]
Thanks for the report: you're assessment is correct. Currently the `gradle-build-action` can only save/restore wrapper downloads that are located in the default location. Are you setting only [`distributionPath`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.wrapper.Wrapper.html#org.gradle.api.tasks.wrapper.Wrapper:distributionPath) or are you also specifying [`distributionBase`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.wrapper.Wrapper.html#org.gradle.api.tasks.wrapper.Wrapper:distributionBase) in `gradle-wrapper.properties`? 

You can configure the action to save/restore more directories inside the Gradle User Home directory, but there's no way to tell the action to save/restore directories outside of the Gradle User Home.

As a workaround:
- If your wrapper dists are located in a different location in Gradle User Home, you could configure `gradle-home-cache-includes` with the extra location.
- If your wrapper dists are located outside of Gradle User Home, then you could setup caching of this directory with `actions/cache`

**Comment by cloudshiftchris** [comment]
Thanks for the info.

Use case is separating our custom Gradle distributions from standard distributions to aid in install / maintenance.

e.g.
`distributionPath=wrapper/my-dists`

...within the GRADLE_USER_HOME distributionBase.

Is there specific cache logic for `wrapper/dists` that wouldn't be applied using a generic `gradle-home-cache-includes`?

**Comment by bigdaz** [comment]
> Is there specific cache logic for wrapper/dists that wouldn't be applied using a generic gradle-home-cache-includes?

Yes, the Gradle wrapper distributions are saved and restored independently of the rest of the Gradle User Home, with the aim to increase sharing between different projects/builds using the same Gradle version.

Another option you have to workaround this limitation is to fork the action and add your custom distribution base here: https://github.com/gradle/gradle-build-action/blob/main/src/cache-extract-entries.ts#LL341C32-L341C32. Note that you'll need to run `npm run build` to regenerate the transpiled sources.
